diff --git a/node_modules/@mswjs/interceptors/lib/node/interceptors/fetch/index.mjs b/node_modules/@mswjs/interceptors/lib/node/interceptors/fetch/index.mjs
index 717678e..6790eb0 100644
--- a/node_modules/@mswjs/interceptors/lib/node/interceptors/fetch/index.mjs
+++ b/node_modules/@mswjs/interceptors/lib/node/interceptors/fetch/index.mjs
@@ -122,7 +122,7 @@ var _FetchInterceptor = class extends Interceptor {
         return mockedResponse;
       }
       this.logger.info("no mocked response received!");
-      return pureFetch(request).then((response) => {
+      return pureFetch(resolvedInput, init).then((response) => {
         const responseClone = response.clone();
         this.logger.info("original fetch performed", responseClone);
         this.emitter.emit("response", {
