diff --git a/index.js b/index.js
index b9f34d5991077fe9fadc7d9187a7225db7e58fca..d22419d72a525a49fc3151321d5889e3441c5a5e 100644
--- a/index.js
+++ b/index.js
@@ -13,7 +13,11 @@
  */
 
 var db = require('mime-db')
-var extname = require('path').extname
+// var extname = require('path').extname
+const extname = function (path) {
+  let index = path.lastIndexOf(".");
+  return index < 0 ? "" : path.substr(index);
+};
 
 /**
  * Module variables.