diff --git a/node_modules/msw/lib/node/index.mjs b/node_modules/msw/lib/node/index.mjs
index b283cc8..9ab2665 100644
--- a/node_modules/msw/lib/node/index.mjs
+++ b/node_modules/msw/lib/node/index.mjs
@@ -1,8 +1,6 @@
 // src/node/SetupServerApi.ts
 import { AsyncLocalStorage } from "node:async_hooks";
-import { ClientRequestInterceptor } from "@mswjs/interceptors/ClientRequest";
 import { XMLHttpRequestInterceptor } from "@mswjs/interceptors/XMLHttpRequest";
-import { FetchInterceptor } from "@mswjs/interceptors/fetch";
 
 // src/node/SetupServerCommonApi.ts
 import { invariant } from "outvariant";
@@ -110,7 +108,7 @@ var AsyncHandlersController = class {
 var SetupServerApi = class extends SetupServerCommonApi {
   constructor(handlers) {
     super(
-      [ClientRequestInterceptor, XMLHttpRequestInterceptor, FetchInterceptor],
+      [XMLHttpRequestInterceptor],
       handlers
     );
     this.handlersController = new AsyncHandlersController(handlers);
